hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/submission/SolvencyPrecheck.java

```diff
@@ -101,14 +101,52 @@ private TxnValidityAndFeeReq assess(SignedTxnAccessor accessor, boolean includeS
             return new TxnValidityAndFeeReq(sigsStatus);
         }
 
-        if (feeExemptions.hasExemptPayer(accessor)) {
+        if (hasExemptPayer(accessor)) {
             return VERIFIED_EXEMPT;
         }
 
         return solvencyOfVerifiedPayer(accessor, includeSvcFee);
     }
 
-    private TxnValidityAndFeeReq solvencyOfVerifiedPayer(SignedTxnAccessor accessor, boolean includeSvcFee) {
+    /**
+     * Checks if the payer account is exempt from paying fees. Public for now to
+     * support a thin adapter to use in {@code IngestChecker}.
+     *
+     * @param accessor the accessor for the transaction
+     * @return whether the payer account is exempt from paying fees
+     */
+    public boolean hasExemptPayer(final SignedTxnAccessor accessor) {
+        return feeExemptions.hasExemptPayer(accessor);
+    }
+
+    /**
+     * Checks if the payer account is valid. Public for now to support a thin
+     * adapter to use in {@code IngestChecker}.
+     *
+     * @param payerNum the payer account number
+     * @return the status of the payer account
+     */
+    public ResponseCodeEnum payerAccountStatus(final EntityNum payerNum) {
+        return queryableAccountStatus(payerNum, accounts.get());
+    }
+
```

Comment:
Probably add TODO : that these will be moved to different classes ?

Reply:
Done!